packages/main/src/plugin/index.ts

```diff
@@ -1075,6 +1076,40 @@ export class PluginSystem {
         return containerProviderRegistry.resolveShortnameImage(providerContainerConnectionInfo, shortName);
       },
     );
+
+    commandRegistry.registerCommand(
+      'pullImage',
+      async (
+        providerContainerConnectionInfo: ProviderContainerConnectionInfo,
+        imageName: string,
+        callbackId: number,
+        platform?: string,
+        taskAction?: TaskAction,
```

Comment:
Do we want to give extensions the option to define a task action? and if so, we'll either have to expose TaskAction in the API or I can break this parameter into optional taskActionName and taskActionExecute parameters, and only if both are provided, we can create a task action inside the command